File: SecondYear/Serve-er_SarveshBaranwal_2226cseAI&ml1045_2/Week_2/countDigits.cpp
int countDigits(int n){
	// Write your code here.
	int count,b,c;
	c=n;
	count=0;
	while(n){
		b=n%10;
		if (b>0){
		if(c%b==0){
        count+=1;
		}
		}
        n=n/10;
		}
	return count;
}

File: SecondYear/aayushraghav93_AayushRaghav_2226CSE(AI)1108_2/Week_2/Count_Digits.java
public class Count_Digits {
    public static int countDigits(int n){
        // Write your code here.
        String nStr = Integer.toString(n);
        int count = 0;

        for (int i = 0; i < nStr.length(); i++) {
            char digitChar = nStr.charAt(i);
            int digit = Character.getNumericValue(digitChar);

            if (digit != 0 && n % digit == 0) {
                count++;
            }
        }

        return count;
    }
}